### DISCUSSION TITLE
How to trigger 'isDirty' when we retrieve the same value when we do reset defaultvalues

### DISCUSSION THREAD

**Discussion by imadegautama** [react-hook-form/react-hook-form#11479] [Q&A]
for example : 

the defaultValues look like this :

{
  name: ""
  isProgrammer : false
}

then we do reset with data like this ({name : "Tama", isProgrammer : false})

the use form will treat isProgrammer isDirty Status as false, but i want the isDirty set to true because is not possible to have value other than true or false. Additionally i use mui as the the input if i use null as the defaultValue mui will complain when the null change. 

Sorry if my english bad, i am looking for the answer asap because i use this for my work project thankyou.

**Comment by Moshyfawn** [comment]
Does [`keepDirtyValues`](https://react-hook-form.com/docs/useform/reset) solve your use-case?

```js
reset(
  { name: "Tama", isProgrammer: false },
  { keepDirtyValues: true }
)
```

**Reply by imadegautama** [reply]
Nope, i had tried before. I think its because they way of useForm work they compared the defaultValue and the current value. Do you have any suggestion for this case

**Reply by imadegautama** [reply]
Can you transfer this to an issue maybe @Moshyfawn
